src/main/scala/coupledL2/tl2chi/MSHR.scala

```diff
@@ -228,7 +231,7 @@ class MSHR(implicit p: Parameters) extends TL2CHIL2Module with HasCHIOpcodes {
       isSnpOnceFwd(req_chiOpcode) ||
     (dirResult.hit && !meta.dirty && meta.state =/= BRANCH || hitWriteEvict) &&
       isSnpOnce(req_chiOpcode)
-  val doRespData = doRespData_dirty || doRespData_retToSrc_fwd || doRespData_retToSrc_nonFwd || doRespData_once
+  val doRespData = (doRespData_dirty || doRespData_retToSrc_fwd || doRespData_retToSrc_nonFwd || doRespData_once) && !denied
```

Comment:
Consider use a signal ```neverRespData``` to suppress ```doRespData```:
```
val neverRespData = denied
...
```

And, according to MainPipe, ```doRespData``` should be named as ```shouldRespData``` here, but no need to modify all those in this PR.